snapshot/doc/user-guide.md

```diff
@@ -0,0 +1,112 @@
+
+# Introduction
+
+Many storage systems provide the ability to create "snapshots" of a persistent volume to protect against data loss. The external snapshot controller and provisioner provide means to use the feature in Kubernetes cluster and handle the volume snapshots through Kubernetes API.
+
+# Features
+
+* Create snapshot of a `PersistentVolume` bound to a `PersistentVolumeClaim`
+* List existing `VolumeSnapshots`
+* Delete existing `VolumeSnapshot`
+* Create a new `PersistentVolume` from an existing `VolumeSnapshot`
+* Supported `PersistentVolume` [types](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes):
+    * Amazon EBS
+    * GCE PD
+    * HostPath
+
+# Lifecycle of a Volume Snapshot and Volume Snapshot Data
+
+## Prerequisites
+Prerequisites for using the snapshotting features are described in sections "Persistent Volume Claim and Persistent Volume" and "Snapshot Promoter".
+
+### Persistent Volume Claim and Persistent Volume
+The user already created a Persistent Volume Claim that is bound to a Persistent Volume. The Persistent Volume type must be one of the snaphot supported Persistent Volume types.
+
+### Snapshot Promoter
+An admin created a Storage Class like the one shown below:
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+```
+Such Storage Class is necessary for restoring a Persistent Volume from already created Volume Snapshot and Volume Snapshot Data.
+
+## Creating
+Each `VolumeSnapshot` contains a spec and status, which is the specification and status of the Volume Snapshot.
+```yaml
+apiVersion: volume-snapshot-data.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: snapshot-demo
+spec:
+  persistentVolumeClaimName: ebs-pvc
+```
+
+* `persistentVolumeClaimName`: name of the Persistent Volume Claim that is bound to a Persistent Volume. This particular Persistent Volume will be snapshotted.
+
+Volume Snapshot Data is automatically created based on the Volume Snapshot. Relationship between Volume Snapshot and Volume Snapshot Data is similar to the relationship between Persistent Volume Claim and Persistent Volume.
+
+Depending on the Persistent Volume type the operation might go through several phases which are reflected by the Volume Snapshot status:
+1. The Volume Snapshot Data is created and is bound to the Volume Snapshot.
+2. The controller starts the snapshot operation: the snapshotted Persistent Volume might need to be frozen and the applications paused.
+3. The storage system finishes creating the snapshot (the snapshot is "cut") and the snapshotted Persistent Volume might return to normal operation. The snapshot itself is not ready yet. The last status condition is of `Created` type with status value "True"
```

Comment:
In the current code, the status condition should be of "Pending" type with status value "True". 

https://github.com/kubernetes-incubator/external-storage/blob/master/snapshot/pkg/apis/crd/v1/types.go#L58